google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/JsonToProtoMessage.java

```diff
@@ -221,11 +221,22 @@ private static DynamicMessage convertJsonToProtoMessageImpl(
                   + ")");
         }
       }
-      if (!field.isRepeated()) {
-        fillField(protoMsg, field, fieldSchema, json, jsonName, currentScope, ignoreUnknownFields);
-      } else {
-        fillRepeatedField(
-            protoMsg, field, fieldSchema, json, jsonName, currentScope, ignoreUnknownFields);
+      try {
+        if (!field.isRepeated()) {
+          fillField(
+              protoMsg, field, fieldSchema, json, jsonName, currentScope, ignoreUnknownFields);
+        } else {
+          fillRepeatedField(
+              protoMsg, field, fieldSchema, json, jsonName, currentScope, ignoreUnknownFields);
+        }
+      } catch (IllegalArgumentException ex) {
```

Comment:
Should we instead create our own exception type here?

Then we can say "catch (OurException) { throw } catch (Exception) {wrap with OurException}", and IIUC this would let us remove all the try-catches inside fillField since this would add the field name to all errors.

Reply:
Good idea, done. The only problem is maybe the info will be duplicated, but it is a cleaner solution.